atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -525,6 +613,29 @@ public void markTableInvolved(TableReference tableRef) {
                 raw -> getWithPostFilteringSync(tableRef, raw, Value.GET_VALUE));
     }
 
+    private Iterator<Map.Entry<Cell, Value>> getRowColumnRangePostFilteredWithoutSorting(
+            TableReference tableRef, Iterator<Map.Entry<Cell, Value>> iterator, int batchHint) {
+        return Iterators.concat(Iterators.transform(Iterators.partition(iterator, batchHint), batch -> {
+            Map<Cell, Value> raw = validateBatch(tableRef, batch);
+            if (raw.isEmpty()) {
+                return Collections.emptyIterator();
+            }
+            Map<Cell, Value> postFiltered = ImmutableMap.copyOf(getWithPostFilteringSync(tableRef, raw, x -> x));
+            return postFiltered.entrySet().iterator();
+        }));
+    }
+
+    private Map<Cell, Value> validateBatch(
+            TableReference tableRef,
+            @org.checkerframework.checker.nullness.qual.Nullable List<Map.Entry<Cell, Value>> batch) {
+        ImmutableMap.Builder<Cell, Value> rawBuilder = ImmutableMap.builder();
+        batch.forEach(rawBuilder::put);
+        Map<Cell, Value> raw = rawBuilder.build();
+
+        validatePreCommitRequirementsOnReadIfNecessary(tableRef, getStartTimestamp());
+        return raw;
+    }
```

Comment:
```suggestion 
    private Map<Cell, Value> validateBatch(
            TableReference tableRef,
            List<Map.Entry<Cell, Value>> batch) {
        validatePreCommitRequirementsOnReadIfNecessary(tableRef, getStartTimestamp());
        return ImmutableMap.copyOf(batch);
    }
```

I _think_ this is equivalent 😅